### PR TITLE
Added support for enviroment variables

### PR DESCRIPTION
**Pull request by nathan-hyan**
## Summary

Added React Native Config and installed pod configs

### Android

image

### iOS

image

## Trello Card

https://trello.com/c/vGxlencm

### CODE REVIEW

**Review by sfernandez11 — CHANGES_REQUESTED** [review]
Great but you forgot [Android extra step](https://github.com/luggit/react-native-config#extra-step-for-android). Did you test it on both platforms?

**Comment by nathan-hyan** [comment]
> Great but you forgot [Android extra step](https://github.com/luggit/react-native-config#extra-step-for-android). Did you test it on both platforms?
> 
> Also, remove `wbooks/package-lock.json`

Found out that there's an easier way of linking the react-native-config package to Android. I'll leave the link here >> https://medium.com/@enzzorp/variables-de-entorno-en-react-native-android-7eaf1eb6114c

**Comment by nathan-hyan** [comment]
Also, deleted package-lock.json

File: wbooks/android/app/src/main/java/com/wbooks/MainApplication.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import android.content.Context;
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactApplication;
+import com.lugg.ReactNativeConfig.ReactNativeConfigPackage;
```
Comment (sfernandez11) [inline wbooks/android/app/src/main/java/com/wbooks/MainApplication.java:7]: No!

File: wbooks/android/settings.gradle [hunk 2]
```diff
@@ -1,3 +1,5 @@
 rootProject.name = 'wbooks'
+include ':react-native-config'
+project(':react-native-config').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-config/android')
```
Comment (sfernandez11) [inline wbooks/android/settings.gradle:3]: No, autolink is deprecated for RN > 0.60. Be careful with this!

File: wbooks/ios/Podfile [hunk 3]
```diff
@@ -8,6 +8,8 @@ target 'wbooks' do
 
   use_react_native!(:path => config["reactNativePath"])
 
+  pod 'react-native-config', :path => '../node_modules/react-native-config'
+
```
Comment (sfernandez11) [inline wbooks/ios/Podfile:12]: Same, not necessary

File: wbooks/src/config/api.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { create } from 'apisauce';
+import Config from 'react-native-config';
+
+const api = create({
+  baseURL: Config.API_BASE_URL,
+  timeout: 5000
+});
+
```
Comment (sfernandez11) [inline wbooks/src/config/api.ts:8]: You forgot this Acceptance Criteria: 
image

**Review by sfernandez11 — CHANGES_REQUESTED** [review]

File: wbooks/src/config/api.ts (hunk 4 above)
Comment (sfernandez11) [reply wbooks/src/config/api.ts:8]: https://github.com/infinitered/reactotron-apisauce#configuring

File: wbooks/android/settings.gradle (hunk 2 above)
Comment (nathan-hyan) [reply wbooks/android/settings.gradle:3]: image

But this is listed on the repository readme, listed under manual link.

File: wbooks/android/app/src/main/java/com/wbooks/MainApplication.java (hunk 1 above)
Comment (nathan-hyan) [reply wbooks/android/app/src/main/java/com/wbooks/MainApplication.java:7]: image

Listed in the library's repository as manual link for android

Comment (sfernandez11) [reply wbooks/android/app/src/main/java/com/wbooks/MainApplication.java:7]: We are not using the manual link, RN has the autolink feature for 0.60+ version

File: wbooks/android/settings.gradle (hunk 2 above)
Comment (sfernandez11) [reply wbooks/android/settings.gradle:3]: We are not using the manual link, RN has the autolink feature for 0.60+ version

**Comment by sfernandez11** [comment]
Check this [Extra step](https://github.com/luggit/react-native-config#extra-step-for-android), I never talk about "Manual link"

File: wbooks/.gitignore [hunk 5]
```diff
@@ -69,3 +69,5 @@ buck-out/
 #
 .jest/
 coverage/
+
+./package-lock.json
```
Comment (sfernandez11) [inline wbooks/.gitignore:73]: Why this? This is not correct, remove it

**Review by sfernandez11 — APPROVED** [review]
🥇

**Pull request merged**